### ISSUE TITLE
Subscriptions Widget: "Optional text" field allows invalid HTML output

### ISSUE THREAD

**Issue by aheckler** [Automattic/jetpack#1298]
To reproduce:

1- Enable Subscriptions module.
2- Add Blog Subscriptions widget to your sidebar.
3- In the "Optional text to display to your readers" field, paste this:

``` html
<ul>
    <li><a href="www.google.com">Link goes here</a></li>
    <li><a href="www.google.com">Link goes here</a></li>
    <li><a href="www.google.com">Link goes here</a></li>
    <li><a href="www.google.com">Link goes here</a></li>
</ul>
```

4- Save the widget and go to your site.
5- View source (don't use Chrome inspector) and observe this HTML:

``` html
<p id="subscribe-text">
    <ul>
        <li><a href="www.google.com">Link goes here</a></li>
        <li><a href="www.google.com">Link goes here</a></li>
        <li><a href="www.google.com">Link goes here</a></li>
        <li><a href="www.google.com">Link goes here</a></li>
    </ul>
</p>
```

According to [the HTML5 spec](http://www.w3.org/TR/html5/grouping-content.html#the-p-element) though, `ul`s (and `ol`s) cannot go inside `p`s.

Reported in 1992335-t for accessibility reasons: "some assistive technology is scarily finicky
about valid HTML".

EDIT: Not reproducible on WP.com by the way.
